woocommerce/woocommerce-ios
# [Mobile Payments] Show spinner while awaiting reader readiness

Closes: #8079


## Description


The `Reader is ready` screen used to show immediately when you tap `Card` on the Payment Method screen, to start the card payment, when a card reader is connected, or immediately after a reader is connected.

This led to a slightly jarring payment experience for the merchant and their customers, as we prompted the merchant to ask their customer to tap their card before the reader is actually ready. There are network-bound operations which happen in this state, so the delay could be tens of seconds in a bad case. On fast Wifi, it's still noticeable at 1-2 seconds of delay

In this PR, we add a `Getting ready to collect payment` screen with a spinner at this point, and then show the `Reader is ready` screen when the Payment Intent has been returned. 

## Testing instructions


Navigate to `Menu > Payments > Collect Payment`
Enter an amount, tap `Next`, then `Take Payment`.
Be ready to tap your card immediately on seeing `Reader is ready` on the phone
Select `Card` as the payment method.
Observe that a spinner shows for a few seconds.
Observe that `Reader is ready` then shows on the phone, at the same time as the green light goes steady on the M2 reader.
Observe that the card can be read immediately.

To see this more clearly, set a breakpoint in Proxyman/Charles on `https://api.stripe.com/v1/payment_intents`, and observe that `Reader is ready` displays before that returns, but you can't tap your card until after you let the breakpoint run.

N.B. This affects the Interac refund flow as well, so Interac payment and refunds should be tested. Try both with and without a connected reader.

I noticed that the `Tap/Swipe/Insert` card alert would not work in the current code, and logged #8116 to fix this. This is not a regression of this PR.

## Screenshots


https://user-images.githubusercontent.com/2472348/201758295-b77d63d4-db75-4e27-aad1-d45bc60741a6.mp4

N.B. ignore the tipping screen: it's leftover from the WisePad configuration I added in my HACK week project and won't show for anyone else.

---
